### PR TITLE
feat: add react `Card` component

### PR DESCRIPTION
**Pull request by hamza14khan**
- Addition of react `Card` component (horizontal, vertical and icon).
- Update associated documentation.
- Documentation files prefix updated.

Testing on Storybook:
Screenshot 2024-10-11 at 16 56 22

Testing on Example application:
Screenshot 2024-10-11 at 16 56 08

### CODE REVIEW

File: packages/react/ds/src/card/card.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            {cloneElement(icon, {
+              ariaLabel: undefined,
+              ariaHidden: true,
+            })}
+          </a>
+        </div>
+      )}
+      <div className="gi-card-content">
+        <div className="gi-card-heading">
+          {href ? (
```
Comment (pregno) [inline packages/react/ds/src/card/card.tsx:49]: Move this ternary operator inside the `Heading`, something like:
```
<Heading as="h5" customClasses="!gi-mt-0 !gi-mb-0">
   {href ? <Link href={href}>{title}</Link> : title} 
</Heading>
```

**Review by pregno — CHANGES_REQUESTED** [review]
Nice work, just a minor change.

**Comment by BogdanDanilescu** [comment]
Minor lint issues
image

File: packages/react/ds/src/card/card.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        type: { summary: 'Array<{ href: string, text: string }>' },
+      },
+    },
+  },
+} satisfies Meta<typeof Card>;
+
+export default meta;
+type Story = StoryObj<typeof meta>;
+
+// Default card example
```
[the remark this answers is not in the document]
Comment (hamza14khan) [reply packages/react/ds/src/card/card.stories.tsx:73]: Not at all. Removed 👍

**Review by pregno — APPROVED** [review]
Wait for Lexa review before merge

File: packages/react/ds/src/card/card.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  href: string;
+  text: string;
+};
+
+export type CardProps = {
+  type: 'vertical' | 'horizontal';
+  title?: string;
+  href?: string;
+  img?: string;
+  icon?: ReactElement;
```
Comment (BogdanDanilescu) [inline packages/react/ds/src/card/card.tsx:16]: can we do here `ReactElement<IconPropTypes>` ? it ensures we are passing the right props

**Review by BogdanDanilescu — APPROVED** [review]
LGTM

**Pull request merged**